src/ua.c

```diff
@@ -493,23 +494,31 @@ static bool require_handler(const struct sip_hdr *hdr,
 }
 
 
-static int sdp_af_hint(struct mbuf *mb)
+static int sdp_originator(struct mbuf *mb, int *af, struct sa *sa)
 {
-	struct pl af;
+	struct pl pl1, pl2;
+	char *addr;
 	int err;
 
+	*af = AF_UNSPEC;
 	err = re_regex((char *)mbuf_buf(mb), mbuf_get_left(mb),
-		       "IN IP[46]+", &af);
+		       "o=[^ ]* [^ ]* [^ ]* IN IP[46]+ [^ \r\n]+",
+		       NULL, NULL, NULL, &pl1, &pl2);
```

Comment:
I am wondering if the o= field should not be used for any logic.
I think the c= field should be used instead ?